app/src/main/kotlin/com/wire/android/ui/home/conversations/sendmessage/SendMessageViewModel.kt

```diff
@@ -273,10 +298,14 @@ class SendMessageViewModel @Inject constructor(
     private fun Either<CoreFailure, Unit>.handleLegalHoldFailureAfterSendingMessage(conversationId: ConversationId) =
         onFailure { it.handleLegalHoldFailureAfterSendingMessage(conversationId) }
 
-    private fun ScheduleNewAssetMessageResult.handleLegalHoldFailureAfterSendingMessage(conversationId: ConversationId) = also {
+    private fun ScheduleNewAssetMessageResult.handleLegalHoldFailureAfterSendingMessage(conversationId: ConversationId) = let {
         if (it is ScheduleNewAssetMessageResult.Failure) {
             it.coreFailure.handleLegalHoldFailureAfterSendingMessage(conversationId)
         }
+        when (this) {
+            is ScheduleNewAssetMessageResult.Failure -> Either.Left(coreFailure)
+            is ScheduleNewAssetMessageResult.Success -> Either.Right(Unit)
+        }
```

Comment:
doesn't the if above cancels the execution of the Failure inside `when`? 🤔

Reply:
I thought `it.coreFailure.handleLegalHoldFailureAfterSendingMessage(conversationId)` was returning something, but its not so the flow is fine 😓